src/backends/torch/torchdataset.cc

```diff
@@ -119,11 +119,23 @@ namespace dd
   }
 
   void TorchDataset::image_to_stringstream(const cv::Mat &img,
-                                           std::ostringstream &dstream)
+                                           std::ostringstream &dstream,
+                                           const bool &lossless)
   {
     std::vector<uint8_t> buffer;
-    std::vector<int> param = { cv::IMWRITE_JPEG_QUALITY, 100 };
-    cv::imencode(".jpg", img, buffer, param);
+    std::vector<int> param;
+    std::string ext;
+    if (compression)
```

Comment:
don't forget to change it here :)
and lossless == png so the branches need to be exchanged